frontend/js/src/common/brainzplayer/AppleMusicPlayer.tsx

```diff
@@ -209,38 +207,44 @@ export default class AppleMusicPlayer
         `/v1/catalog/{{storefrontId}}/search`,
         { term: searchTerm, types: "songs" }
       );
+      const releaseName = _get(listen, "track_metadata.release_name");
       // Remove accents from both the search term and the API results
       const trackNameWithoutAccents = deburr(trackName);
+      const releaseNameWithoutAccents = deburr(releaseName);
       const candidateMatches = response?.data?.results?.songs?.data.map(
         (candidate) => ({
           ...candidate,
           attributes: {
             ...candidate.attributes,
             name: deburr(candidate.attributes.name),
+            albumName: deburr(candidate.attributes.albumName),
           },
         })
       );
-      // Check if the first API result is a match
-      if (
-        new RegExp(escapeRegExp(trackNameWithoutAccents), "igu").test(
-          candidateMatches?.[0]?.attributes.name
-        )
-      ) {
-        // First result matches track title, assume it's the correct result
-        await this.playAppleMusicId(candidateMatches[0].id);
-        return;
-      }
-      // Fallback to best fuzzy match based on track title
-      const fruzzyMatches = fuzzysort.go(
+      const fuzzyMatches = fuzzysort.go(
         trackNameWithoutAccents,
         candidateMatches,
         {
           key: "attributes.name",
-          limit: 1,
         }
       );
-      if (fruzzyMatches[0]) {
-        await this.playAppleMusicId(fruzzyMatches[0].obj.id);
+
+      const matchWithAlbum = fuzzyMatches.find((match) => {
```

Comment:
This should probably only be run if releaseName is truthy (I believe it is optional in the listen metadata so could potentially be undefined/null).

I'm also reading the documentation for fuzzysort, and I wonder if instead of a secondary pass, this couldn't all be achieved in a single pass (when calling fuzzysort.go above )by:
- using both keys `attributes.name` and `attribute.albumName`
- passing a custom `scoreFn` that boosts results that match the albumName

(See 'advanced usage' section https://github.com/farzher/fuzzysort?tab=readme-ov-file#advanced-usage )